zerver/tests/test_outgoing_webhook_system.py

```diff
@@ -210,6 +211,43 @@ def deactivate_bot_first(*args: Any) -> None:
 
         self.assertEqual(bot_owner_notification.recipient_id, bot_user.bot_owner.recipient_id)
 
+    def test_notify_bot_owner_with_deactivated_owner(self) -> None:
+        bot_user = self.example_user("outgoing_webhook_bot")
+        mock_event = self.mock_event(bot_user)
+        service_handler = GenericOutgoingWebhookService("token", bot_user, "service")
+
+        def deactivate_bot_owner_first(*args: Any) -> None:
+            bot_user = get_user_profile_by_id(args[0]["user_profile_id"])
+            assert bot_user.bot_owner is not None
+            do_deactivate_user(bot_user.bot_owner, acting_user=None)
+            notify_bot_owner(*args)
+
+        expected_fail = mock.patch(
+            "zerver.lib.outgoing_webhook.notify_bot_owner", side_effect=deactivate_bot_owner_first
+        )
+
+        with mock.patch.object(
+            service_handler, "session"
+        ) as session, expected_fail as mock_fail, self.assertLogs(level="WARNING") as m:
+            session.post.return_value = ResponseMock(408)
```

Comment:
Can you explain what's happening a bit more here?  E.g. why is it a 408 error?

It might be cleaner to just have this be the case where both the bot and the bot owner are deactivated, since that's what is most likely to happen in practice (as deactivating a user cascades to deactivating all their bots).

Reply:
408 is timeout error.